### PR TITLE
Add rails 6 / 6.1 + ruby 3 support

### PR DESCRIPTION
**Pull request by chaadow**
- Remove last ruby 3 warnings and make the gem totally compatible with
  ruby 3
- Latest versions of rails added `#_write_attribute` for performance
  purposes. this commit mirrors the rails change, as without it
  collection methods such as `<<` do not work anymore.
- Add support for rails 6 and 6.1 while removing rails 4.x and 5.x
  from the travis matrix.
  See [Rails maintenance policy](https://guides.rubyonrails.org/maintenance_policy.html)
- Add support for rails master to catch any upcoming breaking change up
  front.
- Prepare for Rails 6.2 breaking change by updating how errors
  are accessed and removing warning. They are now ruby objects.
  see [this](https://api.rubyonrails.org/v6.1.0/classes/ActiveModel/Errors.html)
- Update specs accordingly

### CODE REVIEW

**Comment by chaadow** [comment]
Hey @manuelmeurer happy holidays!

I would like to ask you if it's possible to update the gemspec to make this a major version ( since it contains breaking changes regarding error objects that will be released in rails 6.2). as well as updating the activesupport/activerecord  minimium versions from `4.2` to `6.0`
I think this strategy can work for the future for multiple reasons: 
1- Should any security release be required for previous versions (that support rails 5.x and 4.x) a branch can always be created from the `v4.0.3` tag 👌 
2- We can clean any old Rails 5 code checking ( `if Rails.version.version <= '5.1'` etc.) and make rails 6 the standard since 5.2 will soon be unmaintained.
3- side note, since i've added rails master as part of the travis matrix, it will be easier to catch any errors as quick as possible

thanks!

**Comment by manuelmeurer** [comment]
Hi @chaadow,

Thanks a lot for your contribution!

I am not using this gem in any of my projects anymore, and unfortunately I have a lot of work on my hands with other stuff, so I won't be able to review your changes in the foreseeable future... :( Sorry for that!

What I can offer you is that you can take over maintenance of this gem, if you're interested.

**Comment by chaadow** [comment]
@manuelmeurer Yes i'm interested. I will try my best, thanks for the offer.

**Comment by manuelmeurer** [comment]
@chaadow I can't transfer this repo to your account since you have a repo with the same name (your fork).
Could you delete that? Then I can transfer it.

**Comment by chaadow** [comment]
@manuelmeurer Done, and thanks!

**Comment by manuelmeurer** [comment]
@chaadow I requested the transfer, I think you have to approve it now.

**Comment by manuelmeurer** [comment]
Could you send me your Rubygems.org handle/email to manuel@krautcomputing.com?
Then I can give you access rights to push new versions of the gem.

**Comment by chaadow** [comment]
@manuelmeurer Email sent 👌🏼

File: lib/active_record/acts_as/instance_methods.rb [hunk 1]
```diff
@@ -19,8 +19,8 @@ def changed?
 
       def actable_must_be_valid
         unless acting_as.valid?
-          acting_as.errors.each do |attribute, message|
-            errors.add(attribute, message) unless errors[attribute].include?(message)
+          acting_as.errors.each do |error|
+            errors.add(error.attribute, error.message) unless errors[error.attribute].include?(error.message)
```
Comment (tindron) [inline lib/active_record/acts_as/instance_methods.rb:23]: This change is only compatible with rails 6.1 and above.

Comment (chaadow) [reply lib/active_record/acts_as/instance_methods.rb:23]: No it works on rails 6.0 as well see [here](https://api.rubyonrails.org/v6.0.0/classes/ActiveModel/Errors.html)

My CI runs against 6.0, 6.1 and rails master

Comment (tindron) [reply lib/active_record/acts_as/instance_methods.rb:23]: Your CI is not running against 6.0.  If you look at the build log you can see the 6.0 build is running 6.1.0.

https://github.com/chaadow/active_record-acts_as/blob/master/gemfiles/rails_6.0.gemfile#L6 has a version constraint that is too permissive.  It should be "~> 6.0.0".

Running the specs against 6.0 gives:
```
Failures:

  1) ActiveRecord::Base model with #acts_as called instance errors when validates_actable is set to true combines supermodel and submodel errors
     Failure/Error: errors.add(error.attribute, error.message) unless errors[error.attribute].include?(error.message)

     NoMethodError:
       undefined method `attribute' for :name:Symbol
     # ./lib/active_record/acts_as/instance_methods.rb:23:in `block in actable_must_be_valid'
```

Comment (chaadow) [reply lib/active_record/acts_as/instance_methods.rb:23]: you're right, I'll look into that

Comment (chaadow) [reply lib/active_record/acts_as/instance_methods.rb:23]: fixed in [5.0.1](https://rubygems.org/gems/active_record-acts_as/versions/5.0.1)

**Pull request merged**